config/config.go

```diff
@@ -302,6 +309,16 @@ func LoadReceiver(componentConfig *viper.Viper, typeStr configmodels.Type, fullN
 		return nil, errorUnmarshalError(receiversKeyName, fullName, err)
 	}
 
+	// run the custom configuration validator for the current component
+	// if no custom config validation logic implemented it calls configmodels.Validate
+	validate, err := validator(receiverCfg)
```

Comment:
We should not validate configs during loading. We should call validate in `configmodels.Config.Validate()` and validate all configs there.

https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configmodels/config.go#L57